inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp

```diff
@@ -408,25 +412,97 @@ Engine::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork &network, const std
         }
     }
 
-    // TODO: handle input precision differently - per input and not one per network...
+    auto config = orig_config;
+    CNNNetwork clonedNetwork = InferenceEngine::details::cloneNetwork(network);
+    const auto& lptProp = config.find(InferenceEngine::PluginConfigInternalParams::KEY_LP_TRANSFORMS_MODE);
+    const bool enableLPT = (lptProp != config.end() && lptProp->second == PluginConfigParams::YES) /* enabled in the orig_config*/
+            || Config::LPTransformsMode::On == engConfig.lpTransformsMode /* or already enabled for the plugin */;
+    auto nGraphFunc = clonedNetwork.getFunction();
+    TransformationUpToCPUSpecificOpSet(nGraphFunc, enableLPT);
+
+    // Here the OV perf modes are turned into specific settings (as we need the network for better params selection)
+    const auto& mode = config.find(PluginConfigParams::KEY_PERFORMANCE_HINT);
```

Comment:
Minor: I would prefer to keep PerfHints related logic in separate method

Reply:
good point, will consider refactoring the logic to the dedicated func  just like I did for the GPU:  https://github.com/openvinotoolkit/openvino/pull/6993/files#diff-2068b2d1a649108c988c3ac72a911ecbcffabd68900214582cc8b5d50c6dd251